ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/submissionManager.service.ts

```diff
@@ -194,11 +211,21 @@ export class SubmissionManager implements OnDestroy {
                 });
             }));
 
+        // remove submissions that include failed answers PATCH with 422 error
+        const submissionRemovalOnPatch422Error$: QueueOpObservable = this.answerSubmissionPatch422ErrorSubject.pipe(
+            map((stableIds: string[]) => (queue: ActivityInstanceAnswerSubmission[]) => {
+                return queue.filter(item => {
+                    return !stableIds.includes(item.stableId);
```

Comment:
> "GOOD VALUE" would be removed from the queue and would not be sent to server and I think this would not be right.

It works as expected for now.

https://user-images.githubusercontent.com/7396837/157223554-60cedaa9-d8ef-471f-b32e-a7ec42213127.mov

Reply:
The thing is that in current implementation we just push every submission in the queue, not checking if there is already the same submission with another ('wrong') value (submissionManager.service.ts, line 191). 

However, even if we check and push only a new (non-existing) submission in the queue - we will have at least one submission (with that invalid value) in the queue until we changed the value. 
        
  In this case we have freezing activity (`isPageBusy = true`) till the answer is valid. Because we rely to the condition
 ` this.isAnswerSubmissionInProgress$ = this.pendingAnswerSubmissionQueue$.pipe(map(queue => queue.length > 0), share());`  (submissionManager.service.ts, line 243)

See screenshot:
![scr11](https://user-images.githubusercontent.com/7396837/157225640-4888b55f-157c-40f0-a738-8a1a1c960d4f.png)

Reply:
I don't think I am following your explanation here. Maybe we can talk about it.
The thing that still does not look right to me is that the QueueOp you implemented can potentially remove more than one item from the queue. I think it is correct that the queue gets updated, but my expectation is that it would be just one the one item that caused the error in the first place. And I think it would be the the last item in the queue. Or even more specific, the last item in the queue that has the corresponding question stable id.
Current use of `filter` can potentially remove 2 or more items and I don't think that is what we want. I think we just want to remove one. The last one. But if you think I am misundertanding things, just reach out.

Reply:
Fixed as discussed in c9623f9f44b3a0765edefdae5ee74da9c6c6f706